### ISSUE TITLE
react/no-unused-prop-types false positive in useEffect with destructuring

### ISSUE THREAD

**Issue by RyanAbraham** [yannickcr/eslint-plugin-react#2350]
It seems that destructuring props inside of useEffect causes the `react/no-unused-prop-types` error to be triggered despite the prop being used. In the example below, calling the prop as `prop.loggedIn` instead of first destructuring it will fix the issue and make the error go away.

```
useEffect(() => {
	const { loggedIn, isAdmin } = props;
	.....
}, []);
```
